src/chef_authn.hrl

```diff
@@ -41,11 +41,13 @@
 %% we assume a default value.
 -define(DEFAULT_SERVER_API_VERSION, 0).
 
-%% version 1.2 incorporates the related but slightly different RSA PKCS 1.5 SHA+RSA signing method
--define(SIGNING_VERSIONS, [?SIGNING_VERSION_V1_0,
-                           ?SIGNING_VERSION_V1_1,
-                           ?SIGNING_VERSION_V1_2,
-                           ?SIGNING_VERSION_V1_3]).
+%% This structure defines allowable hashing algorithms for each signing version.
+%% The frist one in the list is considered to be the default if none is
+%% specified
+-define(SIGNING_VERSIONS, [{?SIGNING_VERSION_V1_0, [?SIGNING_ALGORITHM_SHA1]},
```

Comment:
Instead of proplist, can we just have call into a matched fun header in `chef_authn.erl`?  e.g. 
```erlang
signing_protocols_for_version(?SIGNING_VERSION_V1_0) ->
  [?SIGNING_ALGORITHM_SHA1]; 
signing_protocols_for_version(?SIGNING_VERSION_V1_1) -> 
  [?SIGNING_ALGORITHM_SHA1]; 
% etc
```, etc?